app/reconciler/readDbIntoCompareStructure.server.ts

```diff
@@ -0,0 +1,31 @@
+import { get, set } from 'lodash';
+import { getReconciliationData } from '~/models/dns-record.server';
+import { buildDomain } from '../utils';
+
+import type { ReconcilerCompareStructure } from './ReconcilerTypes';
+
+/**
+ * This Fn reads the database `Record` table into a `ReconcilerCompareStructure` type object
+ * to be used for later comparison with Route53
+ */
+const readDbIntoCompareStructure = async (): Promise<ReconcilerCompareStructure> => {
+  const MUTATEDcompareStructure: ReconcilerCompareStructure = {};
+
+  const dnsData = await getReconciliationData();
+
+  // Populate compareStructure with data from the `Record` table
+  dnsData.forEach(({ username, subdomain, type, value }) => {
+    const fqdn = `${buildDomain(username, subdomain)}.`;
+
+    // Get the previous value for fqdn.recordType (or an empty array if missing)
```

Comment:
This could use an example, since we don't store data as `fqdn.recordType` elsewhere, so readers will wonder what's happening if not reading this as a DIFF like I am now with full context.